internal/ethapi/api.go

```diff
@@ -968,22 +968,37 @@ func (api *BlockChainAPI) GetBlockReceipts(ctx context.Context, blockNrOrHash rp
 // if stateDiff is set, all diff will be applied first and then execute the call
 // message.
 type OverrideAccount struct {
-	Nonce     *hexutil.Uint64             `json:"nonce"`
-	Code      *hexutil.Bytes              `json:"code"`
-	Balance   *hexutil.Big                `json:"balance"`
-	State     map[common.Hash]common.Hash `json:"state"`
-	StateDiff map[common.Hash]common.Hash `json:"stateDiff"`
+	Nonce            *hexutil.Uint64             `json:"nonce"`
+	Code             *hexutil.Bytes              `json:"code"`
+	Balance          *hexutil.Big                `json:"balance"`
+	State            map[common.Hash]common.Hash `json:"state"`
+	StateDiff        map[common.Hash]common.Hash `json:"stateDiff"`
+	MovePrecompileTo *common.Address             `json:"movePrecompileToAddress"`
 }
 
 // StateOverride is the collection of overridden accounts.
 type StateOverride map[common.Address]OverrideAccount
 
 // Apply overrides the fields of specified accounts into the given state.
-func (diff *StateOverride) Apply(statedb *state.StateDB) error {
+func (diff *StateOverride) Apply(statedb *state.StateDB, precompiles vm.PrecompiledContracts) error {
 	if diff == nil {
 		return nil
 	}
 	for addr, account := range *diff {
+		p, isPrecompile := precompiles[addr]
```

Comment:
That's a very good catch! it took a good while to think through the problem and your solution.

I believe the edge-case you found is somewhat vague. It's not clear what *should* happen. Let's expand that edge-case and override those addresses with EVM code. Remember this is the main intention behind the precompile move feature:

```json
{
  "0x1": { "code": "0xff", "movePrecompileTo": "0x2" },
  "0x2": { "code": "0xaa", "movePrecompileTo": "0x3" }
}
```

IMO the core issue is that there is 3 "writes" to address "0x2". 1. that P1 is being moved there, 2. It is setting some EVM code, 3. The precompile at that address is moving to 0x3.

We can see how this vagueness plays into the suggested solution. There "0x2" will lose it's EVM code and will still be considered a precompile, i.e. it will not be deleted from the precompile set because it is present in `newPrecompiles`.

What I plan to do is: allow each address to be written to only once. And consider a precompile move to an address a  write as well. There is no good use-case for this anyway. Yes maybe we could have done a precompile swap like 0x1 -> 0x2 && 0x2 -> 0x1 but why would you want that anyway 😄 so in both these cases we return an error instead.